testonly/testing.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testonly
+
+// TestOrBench is an intersection of testing.T and testing.B methods.
+type TestOrBench interface {
```

Comment:
@daviddrysdale do you have an opinion on this idea?

Reply:
It kind of feels like a hack, but it might be that it's a hack that's easier than shifting everything to the 'right' way of doing things.  

In this context, I'd say the 'right' way to do things is to avoid passing around `*testing.T` as much as possible (and avoid storing it in structures); instead, have helper functions that just take normal (non-test) inputs and return `error`s, and have their `Test...` callers handle any `t.Error`ing. )

(Also, as a nit `TestOrBench` isn't really a great name -- it describes the type in terms of how you invented it rather than what it does.  Maybe something like `TestReporter` / `TestController` / `TestManager` would be a better description of what the type *does*.)

Reply:
For archival purposes: Turns out there is a `testing.TB` interface that does the same thing.